src/lua/lua_redis.c

```diff
@@ -1009,6 +1013,56 @@ rspamd_lua_redis_prepare_connection(lua_State *L, int *pcbref, gboolean is_async
 		}
 		lua_pop(L, 1);
 
+		/* TLS options (optional) */
+		lua_pushstring(L, "ssl");
+		lua_gettable(L, -2);
+		if (!!lua_toboolean(L, -1)) {
+			tls_opts.use_tls = 1;
+		}
+		lua_pop(L, 1);
+
+		lua_pushstring(L, "no_ssl_verify");
+		lua_gettable(L, -2);
+		if (!!lua_toboolean(L, -1)) {
+			tls_opts.no_ssl_verify = 1;
+		}
+		lua_pop(L, 1);
+
+		lua_pushstring(L, "ssl_ca");
+		lua_gettable(L, -2);
+		if (lua_type(L, -1) == LUA_TSTRING) {
+			tls_opts.ca_file = lua_tostring(L, -1);
```

Comment:
that won't work: lua_tostring returns ephemeral string, so we must do some proper memory management, as it will cause hard to debug issues.

Reply:
Hopefully duping the values into variables before using them, then g_free-ing them after connecting is sufficient.